Source/ModelGroups/Node_Attribute.cs

```diff
@@ -1,167 +1,136 @@
-                    material.MetallicRoughnessMaterial.BaseColorTexture = new Runtime.Texture();
-                    material.MetallicRoughnessMaterial.BaseColorTexture.Source = req.value;
-                }
+                node.Translation = new Vector3(-2, 0, 0);
+                properties.Add(new Property(PropertyName.Translation, node.Translation));
             }
 
-            foreach (Property property in combo)
+            void SetTranslationY(List<Property> properties, Runtime.Node node)
             {
-                if (property.name == Propertyname.Matrix)
-                {
-                    nodeList[1].Matrix = specialProperties[0].value;
-                }
-                else if (property.name == Propertyname.Translation ||
-                         property.name == Propertyname.Translation_X ||
-                         property.name == Propertyname.Translation_Y ||
-                         property.name == Propertyname.Translation_Z)
-                {
-                    nodeList[1].Translation = property.value;
-                }
-                else if (property.name == Propertyname.Rotation)
-                {
-                    nodeList[1].Rotation = specialProperties[1].value;
-                }
-                else if (property.name == Propertyname.Scale)
-                {
-                    nodeList[1].Scale = property.value;
-                }
+                node.Translation = new Vector3(0, 2, 0);
+                properties.Add(new Property(PropertyName.Translation, node.Translation));
+            }
+
+            void SetTranslationZ(List<Property> properties, Runtime.Node node)
+            {
+                node.Translation = new Vector3(0, 0, -2);
+                properties.Add(new Property(PropertyName.Translation, node.Translation));
+            }
+
+            void SetRotation(List<Property> properties, Runtime.Node node)
+            {
+                var rotation = Quaternion.CreateFromAxisAngle(new Vector3(0f, 1f, 0f), (float)Math.PI);
+                rotation.W = (float)Math.Round(rotation.W);
+                node.Rotation = rotation;
+                properties.Add(new Property(PropertyName.Rotation, rotation));
             }
 
-            // Apply the material to each node
-            foreach (var node in nodeList)
+            void SetScale(List<Property> properties, Runtime.Node node)
             {
-                node.Mesh.MeshPrimitives[0].Material = material;
+                node.Scale = new Vector3(1.2f, 1.2f, 1.2f);
+                properties.Add(new Property(PropertyName.Scale, node.Scale));
             }
 
-            return wrapper;
+            void SetMatrix(List<Property> properties, Runtime.Node node)
+            {
+                var matrixT = Matrix4x4.CreateTranslation(new Vector3(-2, 2, -2));
+                var matrixR = Matrix4x4.CreateFromAxisAngle(new Vector3(0f, 1f, 0f), (float)(Math.PI));
```

Comment:
You can use [CreateRotationY](https://msdn.microsoft.com/en-us/library/dn889116(v=vs.111).aspx) instead

Reply:
Fixed